### PR TITLE
Add support for managing `kube_cluster` resources with tctl and fix tsh for large labels

### PR DESCRIPTION
**Pull request by tigrato**
* Fixes the tsh ASCII table when a large number of labels are present.
* Adds support for managing `kube_cluster` resources with tctl.

### CODE REVIEW

File: tool/tctl/common/resource_command.go [hunk 1]
```diff
@@ -1225,7 +1253,7 @@ func (rc *ResourceCommand) getCollection(ctx context.Context, client auth.Client
 			}
 		}
 		if len(out) == 0 {
-			return nil, trace.NotFound("database server %q not found", rc.ref.Name)
+			return nil, trace.NotFound("kubernetes server %q not found", rc.ref.Name)
```
Comment (jakule) [inline tool/tctl/common/resource_command.go:1256]: Good catch 😅

File: tool/tsh/kube.go [hunk 2]
```diff
@@ -668,6 +670,8 @@ func newKubeLSCommand(parent *kingpin.CmdClause) *kubeLSCommand {
 	c.Flag("format", defaults.FormatFlagDescription(defaults.DefaultFormats...)).Short('f').Default(teleport.Text).EnumVar(&c.format, defaults.DefaultFormats...)
 	c.Flag("all", "List kubernetes clusters from all clusters and proxies.").Short('R').BoolVar(&c.listAll)
 	c.Arg("labels", labelHelp).StringVar(&c.labels)
+	c.Flag("verbose", "Print full list of labels.").Short('v').BoolVar(&c.verbose)
```
Comment (jakule) [inline tool/tsh/kube.go:673]: Why `verbose` prints all labels? Should that option be called more like `--all` or `--all-labels`?

File: tool/tctl/common/collection.go [hunk 3]
```diff
@@ -918,6 +918,37 @@ func (c *kubeServerCollection) writeJSON(w io.Writer) error {
...
+}
+
+func (c *kubeClusterCollection) resources() (r []types.Resource) {
+	for _, resource := range c.clusters {
+		r = append(r, resource)
+	}
+	return r
+}
+
+func (c *kubeClusterCollection) writeText(w io.Writer) error {
```
Comment (jakule) [inline tool/tctl/common/collection.go:933]: nit: Can you add a comment with some example output? It makes it easier to reason about the output when you can see what it should look like.

File: tool/tsh/kube.go [hunk 4]
```diff
@@ -732,19 +736,30 @@ func (c *kubeLSCommand) run(cf *CLIConf) error {
...
 	switch format {
 	case teleport.Text, "":
-		var t asciitable.Table
-		if cf.Quiet {
-			t = asciitable.MakeHeadlessTable(2)
-		} else {
-			t = asciitable.MakeTable([]string{"Kube Cluster Name", "Labels", "Selected"})
-		}
+		var (
+			t       asciitable.Table
```
Comment (jakule) [inline tool/tsh/kube.go:740]: Can you add a test to cover that?

File: tool/tctl/common/collection.go (hunk 3 above)
Comment (jakule) [inline tool/tctl/common/collection.go:933]: Can you add a test?

File: tool/tsh/kube.go (hunk 2 above)
Comment (tigrato) [reply tool/tsh/kube.go]: I followed the same criteria used for nodes, databases and apps:
https://github.com/gravitational/teleport/blob/7208411f8fbc3794397040312d06ac4458c6c24b/tool/tsh/tsh.go#L2127-L2143

File: tool/tctl/common/collection_test.go [hunk 5]
```diff
@@ -0,0 +1,139 @@
+/*
+Copyright 2015-2017 Gravitational, Inc.
```
Comment (jakule) [inline tool/tctl/common/collection_test.go:2]: nit:
```suggestion
Copyright 2022 Gravitational, Inc.
```

File: tool/tsh/kube_test.go [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+	"github.com/gravitational/teleport/lib/service"
+	"github.com/gravitational/teleport/lib/utils"
+	"github.com/stretchr/testify/require"
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+)
+
+func TestListKube(t *testing.T) {
+	lib.SetInsecureDevMode(true)
+	defer lib.SetInsecureDevMode(false)
```
Comment (jakule) [inline tool/tsh/kube_test.go:42]: nit:
```suggestion
	t.Cleanup(func() {lib.SetInsecureDevMode(false)})
```

File: tool/tsh/kube_test.go [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+	"sort"
+	"strings"
+	"testing"
+
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib"
+	"github.com/gravitational/teleport/lib/asciitable"
+	"github.com/gravitational/teleport/lib/service"
+	"github.com/gravitational/teleport/lib/utils"
+	"github.com/stretchr/testify/require"
```
Comment (jakule) [inline tool/tsh/kube_test.go:35]: nit:
```suggestion
	"github.com/gravitational/teleport/lib/utils"
	
	"github.com/stretchr/testify/require"
```

**Review by jakule — APPROVED** [review]

**Review by r0mant — APPROVED** [review]

**Pull request merged**